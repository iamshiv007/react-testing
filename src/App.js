import logo from './logo.svg';
import './App.css';
import { useState } from 'react';

function App() {
  const [data, setData] = useState("");
  const [isClicked, setIsClicked] = useState(false);

  return (
    <div className="App">
      <p>First react app</p>
      <p>This project made for to learn testing in a react application</p>
      <img src={logo} alt="React logo" title="React logo image" />
      <input type="text" placeholder="This is demo input box" id="demoInput" value={data} onChange={(e) => setData(e.target.value + "test")} />
      <button onClick={() => setIsClicked(!isClicked)}>touch me</button>
      <p>{isClicked ? "Button clicked" : "Button not clicked"}</p>
    </div>
  );
}

export default App;
