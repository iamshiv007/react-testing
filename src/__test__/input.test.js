import { fireEvent, render, screen } from "@testing-library/react";
import App from "../App";

beforeEach(() => {
    console.log("hello this is before each tetsing")
})

test("test for input field", () => {
    render(<App />)

    const inputField = screen.getByPlaceholderText(/This is demo input box/)
    const inputFieldByRole = screen.getByRole("textbox")

    expect(inputField).toBeInTheDocument()
    expect(inputFieldByRole).toBeInTheDocument()

    fireEvent.change(inputField, { target: { value: "a" } })

    expect(inputField.value).toBe("atest")
})

test("test for button click", () => {
    render(<App />)
    const button = screen.getByRole("button")

    const notClickedText = screen.getByText("Button not clicked")
    expect(notClickedText).toBeInTheDocument()
    fireEvent.click(button)

    const clickedText = screen.getByText("Button clicked")
    expect(clickedText).toBeInTheDocument()
})

afterEach(() => {
    console.log("Hello this is after each test")
})

test("snapshot for app component", () => {
    const container = render(<App />)
    expect(container).toMatchSnapshot()
})