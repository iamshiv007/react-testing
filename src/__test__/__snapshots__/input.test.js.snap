// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <p>
          Hello, First react app
        </p>
        <p>
          This project made for to learn testing in a react application
        </p>
        <img
          alt="React logo"
          src="logo.svg"
          title="React logo image"
        />
        <input
          id="demoInput"
          placeholder="This is demo input box"
          type="text"
          value=""
        />
        <button>
          touch me
        </button>
        <p>
          Button not clicked
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <p>
        Hello, First react app
      </p>
      <p>
        This project made for to learn testing in a react application
      </p>
      <img
        alt="React logo"
        src="logo.svg"
        title="React logo image"
      />
      <input
        id="demoInput"
        placeholder="This is demo input box"
        type="text"
        value=""
      />
      <button>
        touch me
      </button>
      <p>
        Button not clicked
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot for sum function 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <p>
          First react app
        </p>
        <p>
          This project made for to learn testing in a react application
        </p>
        <img
          alt="React logo"
          src="logo.svg"
          title="React logo image"
        />
        <input
          id="demoInput"
          placeholder="This is demo input box"
          type="text"
          value=""
        />
        <button>
          touch me
        </button>
        <p>
          Button not clicked
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <p>
        First react app
      </p>
      <p>
        This project made for to learn testing in a react application
      </p>
      <img
        alt="React logo"
        src="logo.svg"
        title="React logo image"
      />
      <input
        id="demoInput"
        placeholder="This is demo input box"
        type="text"
        value=""
      />
      <button>
        touch me
      </button>
      <p>
        Button not clicked
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
